api/src/paths/dwc/validate.ts

```diff
@@ -0,0 +1,223 @@
+                          type: 'array',
+                          items: {
+                            type: 'string'
+                          }
+                        },
+                        headerErrors: {
+                          type: 'array',
+                          items: {
+                            type: 'object',
+                            properties: {
+                              type: {
+                                type: 'string'
+                              },
+                              code: {
+                                type: 'string'
+                              },
+                              message: {
+                                type: 'string'
+                              },
+                              col: {
+                                oneOf: [
+                                  {
+                                    type: 'string'
+                                  },
+                                  {
+                                    type: 'number'
+                                  }
+                                ]
+                              }
+                            }
+                          }
+                        },
+                        rowErrors: {
+                          type: 'array',
+                          items: {
+                            type: 'object',
+                            properties: {
+                              type: {
+                                type: 'string'
+                              },
+                              code: {
+                                type: 'string'
+                              },
+                              message: {
+                                type: 'string'
+                              },
+                              row: {
+                                type: 'number'
+                              }
+                            }
+                          }
+                        },
+                        isValid: {
+                          type: 'boolean'
+                        }
+                      }
+                    }
+                  }
+                ]
+              }
+            }
+          }
+        }
+      }
+    },
+    400: {
+      $ref: '#/components/responses/400'
+    },
+    401: {
+      $ref: '#/components/responses/401'
+    },
+    403: {
+      $ref: '#/components/responses/401'
+    },
+    500: {
+      $ref: '#/components/responses/500'
+    },
+    default: {
+      $ref: '#/components/responses/default'
+    }
+  }
+};
+
+function prepDWCArchive(): RequestHandler {
+  return async (req, res, next) => {
+    defaultLog.debug({ label: 'prepDWCArchive', message: 'files.length', files: req?.files?.length });
+
+    if (!req.files || !req.files.length) {
+      // no media objects included
+      throw new HTTP400('Missing upload data');
+    }
+
+    if (req.files.length !== 1) {
```

Comment:
question:

is the reason we accept an array of files instead of just a single file because we eventually wanna support multiple files at once but for now we are only doing the case for one? cause if this is not the case, just wondering if it can be a single file accepted so we don't need a check like this

Reply:
Think we talked about this.  The schema was out of date and should have indicated a single file.  That being said, the schema doesnt actually prevent someone sending multiple files, so this check is still needed for now